hasura/graphql-engine
# allow logging details of a live query polling thread

Currently we don't log any metrics related to the polling that happens in the background to serve live queries. These metrics are captured and summarized internally using datatypes from ekg-core (which can be dumped by `/dev/subscriptions` endpoint). This PR changes this behaviour, the metrics can now be logged by enabling a new log type `livequery-poller-log`. They are not summarized internally anymore. This opens up options to pipe the poller logs into a metric collection system where they can be analyzed further. 




### Description



### Changelog


### Affected components



### Steps to test and verify
Add `livequery-poller-log` to `HASURA_GRAPHQL_ENABLED_LOG_TYPES` and you'll see logs of type: 
```json
{"type":"livequery-poller-log","timestamp":"2020-06-01T20:58:58.565+0530","level":"info","detail":{"poller_id":"4ec1aab0-43e0-4978-bc33-36a4f642ae12","total_time":1.872028e-3,"snapshot_time":2.2264e-5,"batches":[{"pg_execution_time":1.71155e-3,"push_time":9.8706e-5}]}}
```


#### Catalog upgrade

Does this PR change Hasura Catalog version?

## Notes to the reviewer:

Module dependency tree:

![image](https://user-images.githubusercontent.com/6562944/83425645-9c4cc200-a44b-11ea-92bc-4c90f10b10a7.png)
